sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/timeExpressions.scala

```diff
@@ -749,3 +751,288 @@ case class TimeTrunc(unit: Expression, time: Expression)
     )
   }
 }
+
+abstract class IntegralToTimeBase
+  extends UnaryExpression with ExpectsInputTypes with CodegenFallback {
```

Comment:
Can we follow `IntegralToTimestampBase` completely and do not use `CodegenFallback`? These are simple functions and implementing codegen is trivial.

Reply:
Raised a followup [PR](https://github.com/apache/spark/pull/53370)